profile/nodejs.sh

```diff
@@ -6,16 +6,25 @@ export NODE_ENV=${NODE_ENV:-production}
 if [[ -n "$HEROKU_METRICS_URL" ]] && \
    # if we're not on a one-off dyno
    [[ "${DYNO}" != run\.* ]] && \
-   # if the plugin was installed for this node version
-   [[ -d $HOME/.heroku/heroku-nodejs-plugin ]] && \
    # the user has not opted out
    [[ -z "$HEROKU_SKIP_NODE_PLUGIN" ]]; then
 
-  # Don't clobber NODE_OPTIONS if the user has set it, just add the require flag to the end
-  if [[ -z "$NODE_OPTIONS" ]]; then
-      export NODE_OPTIONS="--require $HOME/.heroku/heroku-nodejs-plugin"
-  else
-      export NODE_OPTIONS="${NODE_OPTIONS} --require $HOME/.heroku/heroku-nodejs-plugin"
+  # if the plugin was installed for this node version
+  if [[ -d $HOME/.heroku/heroku-nodejs-plugin ]]; then
+    # Don't clobber NODE_OPTIONS if the user has set it, just add the require flag to the end
+    if [[ -z "$NODE_OPTIONS" ]]; then
+        export NODE_OPTIONS="--require $HOME/.heroku/heroku-nodejs-plugin"
+    else
+        export NODE_OPTIONS="${NODE_OPTIONS} --require $HOME/.heroku/heroku-nodejs-plugin"
+    fi
+  # or if the metrics script was installed
+  elif [[ -d $HOME/.heroku/metrics ]]; then
+    # Don't clobber NODE_OPTIONS if the user has set it, just add the require flag to the end
+    if [[ -z "$NODE_OPTIONS" ]]; then
```

Comment:
This `if`/`else` is a duplicate of the one on line 15. I think the metrics collector/plugin location could be a variable and you could keep it DRY.

Reply:
The `if / else` is not a duplicate.  The new metrics are scoped to only Node v21 for now so the first conditional sets the `NODE_OPTIONS` to use the C++ addon for the earlier Node versions and the second conditional sets it to use the new metrics script.

Reply:
oh, i see.  yeah, i can make that change.